XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs

```diff
@@ -0,0 +1,154 @@
+        {
+            this.InitializeComponent();
+            UpdateTitlebarColor();
+            UpdateButtonText();
+        }
+
+
+        [ComImport, Guid("EECDBF0E-BAE9-4CB6-A68E-9598E1CB57BB"), InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+        private interface IWindowNative
+        {
+            IntPtr WindowHandle { get; }
+        };
+
+        private void SetTitlebar(UIElement titlebar)
+        {
+            var window = App.StartupWindow;
+            if (!window.ExtendsContentIntoTitleBar)
+            {
+                window.ExtendsContentIntoTitleBar = true;
+                window.SetTitleBar(titlebar);
+            }
+            else
+            {
+                window.ExtendsContentIntoTitleBar = false;
+                window.SetTitleBar(null);
+
+            }
+            UpdateButtonText();
+            UpdateTitlebarColor();
+        }
+
+        private void UpdateButtonText()
+        {
+            var window = App.StartupWindow;
+            if (window.ExtendsContentIntoTitleBar)
+            {
+                customTitlebar.Content = "Reset to system Titlebar";
+                defaultTitlebar.Content = "Reset to system Titlebar";
+            }
+            else
+            {
+                customTitlebar.Content = "Set Custom Titlebar";
+                defaultTitlebar.Content = "Set Fallback Custom Titlebar";
+            }
+
+        }
+
+        private void BgColorButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Extract the color of the button that was clicked.
+            Button clickedColor = (Button)sender;
+            var rectangle = (Microsoft.UI.Xaml.Shapes.Rectangle)clickedColor.Content;
+            var color = ((Microsoft.UI.Xaml.Media.SolidColorBrush)rectangle.Fill).Color;
+
+            CurrentBgColor.Background = new SolidColorBrush(color);
+
+            currentBgColor = color;
+            UpdateTitlebarColor();
+
+        }
+
+        private void FgColorButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Extract the color of the button that was clicked.
+            Button clickedColor = (Button)sender;
+            var rectangle = (Microsoft.UI.Xaml.Shapes.Rectangle)clickedColor.Content;
+            var color = ((Microsoft.UI.Xaml.Media.SolidColorBrush)rectangle.Fill).Color;
+
+            CurrentFgColor.Background = new SolidColorBrush(color);
+
+            currentFgColor = color;
+            UpdateTitlebarColor();
+
+        }
+
+
+        private void UpdateTitlebarColor()
+        {
+            var res = Microsoft.UI.Xaml.Application.Current.Resources;
+            res["WindowCaptionBackground"] = currentBgColor;
+            //res["WindowCaptionBackgroundDisabled"] = currentBgColor;
+            res["WindowCaptionForeground"] = currentFgColor;
+            //res["WindowCaptionForegroundDisabled"] = currentFgColor;
+
+            // to trigger repaint tracking task id 38044406

```

Comment:
I assume that this is a bug that we are working around. We should add a comment explaining what the bug is. XCG is used as documentation and readers will likely copy any code that is included here.